cl/corpus_importer/management/commands/troller_bk.py

```diff
@@ -84,61 +160,133 @@ def merge_rss_data(
-                court_id,
-                docket["pacer_case_id"],
-                docket["docket_number"],
-            )
-            document_number = docket["docket_entries"][0]["document_number"]
+        date_filed, time_filed = localize_date_and_time(
+            court_id, docket_entry["date_filed"]
+        )
+
+        d.add_recap_source()
+        if not d.pk:
+            # Docket no exists
             if (
-                document_number
-                and d.docket_entries.filter(
-                    entry_number=document_number
-                ).exists()
+                not docket["pacer_case_id"]
+                and court.jurisdiction != Court.FEDERAL_APPELLATE
             ):
-                # It's an existing docket entry; let's not add it.
+                # Avoid adding the docket if it belongs to a district/bankr
+                # court and doesn't have a pacer_case_id
                 continue
-            if not d.pk:
-                update_docket_metadata(d, docket)
-                dockets_created += 1
-
-            if not d.pacer_case_id:
-                d.pacer_case_id = docket["pacer_case_id"]
-            try:
-                d.save()
-                add_bankruptcy_data_to_docket(d, docket)
-            except (DataError, IntegrityError) as exc:
-                # Trouble. Log and move on
-                logger.warn(
-                    f"Got DataError or IntegrityError while saving docket."
+
+            update_docket_metadata(d, docket)
+            d.pacer_case_id = docket["pacer_case_id"]
+            d.slug = slugify(trunc(best_case_name(d), 75))
+            d.date_last_filing = date_filed
+            if d.docket_number:
+                d.docket_number_core = make_docket_number_core(d.docket_number)
+
+            docket_in_list = unique_dockets.get(docket["docket_number"], None)
+            if not docket_in_list:
+                unique_dockets[docket["docket_number"]] = docket
+                dockets_to_create.append(d)
+
+            existing_entry = des_to_add_no_existing_docket.get(
+                docket["docket_number"]
+            )
+            if existing_entry:
+                des_to_add_no_existing_docket[docket["docket_number"]].append(
+                    docket_entry
                 )
+            else:
+                des_to_add_no_existing_docket[docket["docket_number"]] = [
+                    docket_entry
+                ]
+            continue
 
-            des_returned, rds_created, content_updated = add_docket_entries(
-                d,
-                docket["docket_entries"],
-                do_not_update_existing=True,
+        # Docket exists.
+        des_to_add_existing_docket.append((d.pk, docket_entry))
+        try:
+            d.save()
+        except (DataError, IntegrityError) as exc:
+            # Trouble. Log and move on
+            logger.warn(
+                f"Got DataError or IntegrityError while saving docket."
```

Comment:
Done, previously there was a `continue` at the end of the previous `if`, so the `save` function for existing dockets was not executed. However, I think the else statement is clearer. Additionally, I modified the code to only save the `source` field instead of the entire object, also added here `add_bankruptcy_data_to_docket` so bankr data is added on existing dockets.